daft/dataframe/dataframe.py

```diff
@@ -2818,6 +2841,7 @@ def collect(self, num_preview_rows: Optional[int] = 8) -> "DataFrame":
             DataFrame: DataFrame with materialized results.
         """
         self._materialize_results()
+        self.explain_broadcast()
```

Comment:
Maybe consider making this a decorator for all "execution" methods (`show`, `collect`, `write_parquet` etc)

Reply:
Sounds good. A decorator makes sense here. Will update in the coming commit.

Reply:
Updated to use decorator instead.

Reply:
I will need to add this to more APIs. Do we have an exhaustive list somewhere of all the APIs we would want to add this decorator to?